src/site/content/en/blog/mailru-cwv/index.md

```diff
@@ -0,0 +1,290 @@
+  <figcaption>Web Vitals values before the improvements.</figcaption>
+</figure> 
+
+## Improving CWV
+
+While plenty of guidance exists for improving CWV, every project is unique in its requirements and challenges. For the Mail.ru home page, the following opportunities were identified:
+
+- Implementing placeholders for ad banners to reduce CLS.
+- Use Server-Side Rendering (SSR) to reduce [Largest Contentful Paint (LCP](https://web.dev/lcp/)).
+- Code splitting to reduce LCP and [First Input Delay (FID)](https://web.dev/fid/).
+ 
+## Skeletons for CLS improvement
+
+CLS was one of the worst performing field metrics for the Mail.ru home page. Subsequent profiling of this page in the [Performance panel](https://developer.chrome.com/docs/devtools/evaluate-performance/) of Chrome's DevTools revealed that ads were the source of the problem. To improve layout stability, our team decided to use placeholders to reserve space for ads before they load.
+
+When implementing placeholders, the first step is to determine the dimensions of the content that will replace them. Luckily, the desktop version of Mail.ru home page has strictly documented sizes for ads. After talking with the design team, SVG-animated UI skeletons were used as placeholders as [they reduce the perceived load time of the content](https://uxdesign.cc/what-you-should-know-about-skeleton-screens-a820c45a571a).
+ 
+{% Aside 'caution' %}
+SVG animations are not supported in Internet Explorer 11, consider using CSS if legacy browser support is essential to your application.
+{% endAside %}
+
+
+## The return of SSR
+
+To ease the transition from Fest to Svelte, we incrementally rewrote the existing project rather than start over. By March 2021, we had migrated most of the frontend to Svelte, and eventually brought SSR to our production application after triaging and fixing backend performance issues.
+
+{% Aside %}
+SSR was not used for the entire application, as [it can be an expensive process](https://malloc.fi/performance-cost-of-server-side-rendered-react-node-js) that can delay [Time to First Byte (TTFB)](https://web.dev/ttfb/), which [indirectly affects LCP](https://web.dev/optimize-lcp/#optimize-your-server). We decided to use SSR for crucial content that the user will see on initial render, and offload rendering of hidden content on the client. At the moment, Mail.ru is experimenting with [Islands Architecture](https://jasonformat.com/islands-architecture/) to see if further performance gains can be had.
+{% endAside %}
+
+After implementing SSR, the cause of regression in CLS metrics was discovered that initially went unnoticed: the news block was not inserted at the moment of rendering the first content on the page. There was a delay between the initial painting of the page and the insertion of news. This behaviour resulted in an ad skeleton shift, which worsened CLS metrics.
```

Comment:
```suggestion
After implementing SSR, the cause of regression in CLS metrics was discovered that initially went unnoticed: the news section was not inserted at the moment of rendering the first content on the page. There was a delay between the initial painting of the page markup provided by the server and the insertion of news section on the client. This behaviour resulted in an ad skeleton shift, which worsened CLS metrics.
```

Reply:
This needed to be a bit clearer. Reject it if it's inaccurate, but that's my read on the situation.